### ISSUE TITLE
In a Razor page returning View from OnGet results in a compile error.

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/Mvc#5802]
Having:

```
@page

@functions {
    public IActionResult OnGet()
    {
        Message = "From OnGet";

        return View();
    }
    public string Message { get; set; } = "Default";
}
<p>The message: @Message</p>
```

Results in a compile error:
![image](https://cloud.githubusercontent.com/assets/2008729/22942000/86a8d97a-f29c-11e6-9fd4-e970b77ac239.png)

**Comment by ryanbrandenburg** [comment]
This seems to have been fixed by e11e6b3. I'll go ahead and add a functional test like he did for Redirect though.

**Comment by rynowak** [comment]
Alright let's go 👍
